### PR TITLE
Unit spawning (no art)

### PR DESCRIPTION
**Pull request by JustinWinningham**
Units now spawn correctly based on the map file data

### CODE REVIEW

File: Globals/GLOBAL.gd [hunk 1]
```diff
@@ -64,3 +64,20 @@ func reloadDirectoryFiles(path, destinationArray):
...
+func getUnitFromId(id):
+	var newUnit = null
+	for m in FACTIONCUSTODIAN.factionArray.size():
+		for n in FACTIONCUSTODIAN.factionArray[m].unitRoster.size():
+			if id == FACTIONCUSTODIAN.factionArray[m].unitRoster[n].id:
+				print_debug("found it!")
+				var unitTemplate = FACTIONCUSTODIAN.factionArray[m].unitRoster[n]
+				newUnit = FACTIONCUSTODIAN.factionArray[m].unitRoster[n]
+				newUnit = Unit.new(null)
+				newUnit.copyFrom(unitTemplate)
```
Comment (brooks42) [inline Globals/GLOBAL.gd]: the constructor can just take the unit template right?

File: Content/Maps/testmap/setupData.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
   "OffsetUnitStartPositionTeamA":[
-  {"unitId":1, "xpos": 2, "ypos": 3},
-  {"unitId":1, "xpos": 3, "ypos": 4}
+  {"unitId":"guardsman", "xpos": 2, "ypos": 3},
```
Comment (brooks42) [inline Content/Maps/testmap/setupData.json:3]: what other metadata can we put in this object? I'm imagining we'll use this same json format for base levels and also save files, so we should plan to be able to store things like movement state (moved/fired or not, etc), current HP, fuel/ammo etc.

File: Globals/GLOBAL.gd (hunk 1 above)
Comment (JustinWinningham) [reply Globals/GLOBAL.gd]: the constructor already takes a filepath for preexisting functions, and we can't create multiple constructors...

File: Content/Maps/testmap/setupData.json (hunk 2 above)
Comment (JustinWinningham) [reply Content/Maps/testmap/setupData.json:3]: that's a good point, I didn't think about the idea of damaged units, different states, etc. I will update this when I implement those systems.

**Review by brooks42 — COMMENTED** [review]
Just some quick comments, and I think we should try to clarify the temp/commented out code, if possible. And of course either move the unit template passing into the constructor or have it as a separate factory.

I'm also still not sure I 100% get the `unit` object here, why is the template and the unit the same thing in the code? I thought the template was just the metadata and the unit itself was a separate kind of entity that used the metadata to define itself?

File: Globals/GLOBAL.gd [hunk 3]
```diff
@@ -64,3 +64,18 @@ func reloadDirectoryFiles(path, destinationArray):
 	else:
 		print_debug("Unable to open desired directory: " + path)
 		return !OK
+
+# From the ID present within the map file, find the appropriate unitmetadata, return said metadata object
+func getUnitMetaDataFromId(id):
+	var metaData = null
+	for m in FACTIONCUSTODIAN.factionArray.size():
+		for n in FACTIONCUSTODIAN.factionArray[m].unitRoster.size():
+			if id == FACTIONCUSTODIAN.factionArray[m].unitRoster[n].id:
```
Comment (brooks42) [inline Globals/GLOBAL.gd:73]: I can't believe we can't do this more cleanly lol

File: Scripts/Objects/Unit.gd [hunk 4]
```diff
@@ -25,39 +18,44 @@ var cost = 0
 var defense = 1
 var offense = 1
 var speed = 1
-# TODO: Think about how we want store stats at the data layer
-# need second pass on design to be more exact
+
+var state = 0 #0 = disabled, 1 = can move and attack, 2 = can attack, not move, 3 = can attack, not move
```
Comment (brooks42) [inline Scripts/Objects/Unit.gd:22]: we can do `enum { DISABLED, CAN_MOVE_AND_ATTACK, etc }` right? Would rather have these things compiler-enforced if possible.

**Review by brooks42 — APPROVED** [review]
Approved with some comments, if you want to make a trello card to refactor the stuff and link this PR I'll handle it at some point.

File: Scripts/Objects/Unit.gd (hunk 4 above)
Comment (JustinWinningham) [reply Scripts/Objects/Unit.gd:22]: it's interpreted language so no compiling. But yes we can change it, I tried doing it that way at first but I couldn't figure out the syntax

**Pull request merged**